Address CIA (#30)
WICG/Realms-Initialization-Control: README.md

The conclusion is that not being able to enforce such services at a very specific time (earlier than all other scripts) to all same origin realms of the app (as opposed to its top most realm only), significantly narrows down the reach such services have, which attackers can abuse.

## Security

This section explores this proposal's impact on security from the perspective of the well known [CIA](https://agoric.com/blog/technology/a-taxonomy-of-security-issues) (Confidentiality,Integrity,Availability) security standard.

### Integrity

Reading through the different sections of this proposal makes it clear RIC is a security feature that aspires to support integrity specifically.

> As described in this document, same origin realms are a power that can allow untrusted code that's invited to run within the same context of a given web app to undermine its will to apply its own set of rules and restrictions on its hosting envirnoment.

Reviewer: ```suggestion
As described in this document, same origin realms are a power that can allow untrusted code that's invited to run within the same context of a given web app to undermine its will to apply its own set of rules and restrictions on its hosting envirnoment.
```
```suggestion
As described in this document, same origin realms are a power that can allow untrusted code that's invited to run within the same context of a given web app to undermine the app's own set of rules and restrictions on its hosting envirnoment.
```

> While such rules and restrictions are applicable via JavaScript runtime virtualization, that statement does not hold against the power to form same origin realms, which is untamable.

Reviewer: ```suggestion
While such rules and restrictions are applicable via JavaScript runtime virtualization, that statement does not hold against the power to create same origin realms (e.g. iframes), which is untamable.
```

> RIC aspires to provide web apps the power to tame it, by allowing them to execute their code within such realms before any other hosted code, so that they can apply their rules and restrictions within them too.

Reviewer: ```suggestion
RIC aspires to provide web apps the power to tame it, by allowing them to execute their code within such realms before any other hosted code, so that they can apply their rules and restrictions within these realms too.
```

> Therefore, in this context, apps using JavaScript runtime virtualization to introduce their own set of rules and restrictions can be thought of an act of preserving a high level of integrity.

Reviewer: ```suggestion
Therefore, in this context, apps using JavaScript runtime virtualization to introduce their own set of rules and restrictions can be thought of as preserving a high level of integrity.
```

Since the power to create same origin realms undermine this goal, it by defenition lowers that level of integrity.

Therefore, by providing a way to restore the power to control the creation of such realms to the app, RIC aspires to recover its integrity level.

To summarize, RIC isn't a standalone security feature, but one that compliments the attempts of app builders and security vendors to harden the integrity of web apps.

### Confidentiality

Since RIC isn't a standalone security feature, but one that focuses specifically on complimenting attempts of integrity hardening, it has no aspiration to address confidentiality.

By focusing on hardening the integrity of the app against entities living within it, RIC by definition addresses security concerns that may occur within a single agent (process).

> Therefore, it does not allow apps to defend themselves against types of confidentiality attacks, such as side channeling (e.g. Meltdown and Spectre).

Reviewer: ```suggestion
Therefore, it does not contribute to the apps' ability to defend themselves against the types of confidentiality attacks, such as side channeling (e.g. Meltdown and Spectre).
```

> That is fine - there's still value in providing apps with APIs to harden their level of integrity without addressing other security aspects.

Reviewer: ```suggestion

```
